src/foam/dao/CSVSink.js

```diff
@@ -76,34 +127,95 @@ foam.CLASS({
             Null: function(value) {}
           }, function(value) {
             this.output_(value.toString());
-          })
+        }),
+      javaCode: `
+        switch (value) {
+          case value == null: 
```

Comment:
I don't think this is valid java code?

You'll need to do a series of if () else if () else if (), similar to how JSON Outputter operates.

Reply:
Actually upon further thought we might not need these conditionals at all.  If you add outputCSV() to PropertyInfo, you can just call that for each axiom we want to output rather than doing the formatting here.

Reply:
https://www.w3schools.com/java/java_switch.asp

Reply:
So just thinking out loud here,
So we are creating a default toCSV on properties(js/java) which can be overridden for any type we want unique behavior.
The output( and output_ ) functions, basically convert csv output of each property to string.

So epiphany idea as I understand it: make toCSV/toCSVLabel(for both js and java) a function that returns a string, and then we can rid ourselves of output functions in CSVSink

correct??